js/advanced.js

```diff
@@ -0,0 +1,54 @@
+$(function(){
+  var bulbsRegistry = {}; 
+  var buttonsRegistry = {};
+
+  discoverBulbs();
+  discoverButtons();
+
+  function discoverBulbs() {
+    $(".bulb").each(function(index){
+      bulbsRegistry[$(this).attr("id")] = $(this).css("backgroundColor");
+      turnBulbOff($(this));
```

Comment:
💯 Nice implementation. The only issue is that all of the lights are on when it starts, then they default to black. That is because it sets the classes, then call `turnBulboff()` to set them to black.